distrobuilder/main.go

```diff
@@ -786,3 +786,19 @@ fi
 	os.MkdirAll("/etc/systemd/system-generators", 0755)
 	ioutil.WriteFile("/etc/systemd/system-generators/lxc", []byte(content), 0755)
 }
+
+// addVethUdevRule creates an udev rule that makes eth0 work with NetworkManager
+func addVethUdevRule() {
```

Comment:
It would be preferable to do this in the systemd generator instead, using a /run/udev/rules.d entry if possible too.

The reason for that is the systemd generator is designed to be effectively free to include on any system, making it possible to then convert a system to a LXD VM or container and have things behave right outside of the box.

The approach in this PR would instead cause issues should a LXD container be turned into a VM or physical system and would similarly not work for anyone converting a physical system or a VM to a container.